wordpress-mobile/WordPress-iOS
# Reader: Fix unsatisfiable constraints

Part of #18098

## Description
This resolves a few unsatisfiable constraints errors related to the reader. Errors fixed are:
- Issue in `NoResultsViewController`
    - This was triggered because the main stack view had leading and trailing constraints to superview. And this was conflicting with the width determined by its content.
    - Fixed by removing leading and trailing constraints and adding a center horizontally constraint instead.
    - This change resulted in no visible change.
- Issue in `ReaderPostCardCell`, related to the two overlapping avatar image views
    - Both image views are inside a stack view. The overlapping was implemented by tying the leading edge of the second avatar to the center of the first avatar. This constraint conflicted with constraints added by the stack view for its subviews.
    - Fixed by removing this constraint and instead, setting the spacing to a negative value equal to half the width of the avatar. This works fine because the avatar width is static and set using a constraint.
    - This change resulted in no visible change.
- Issue in `PostListFooterView`
    - This view had a subview, and the constraints on that subview were conflicting with constraints added by the table view.
    - This was fixed by completely removing the problematic subview as it was always hidden either way.
    - This change resulted in no visible change.
- Issue in `ReaderPostCardCell `, related to ghost view of the cell
    - This was caused by a height constraint between the header stack view and one of its subviews.
    - Because of this issue the header stack view always had a height of `0` and was never visible inside the ghost cell. After looking at how the ghost cell used to look when it was implemented (#14485), this seems like a bug to me.
    - Fixed by removing this constraint and configuring the header stack view.

Before | After
-- | --
<img src="https://user-images.githubusercontent.com/25306722/157365138-8e78e369-c439-432e-a8b1-f0df3b01f756.png" width=200> | <img src="https://user-images.githubusercontent.com/25306722/157365125-adabeed0-9cec-4210-90f6-452948cb4e47.png" width=200>

## Testing Instructions
1. Open the reader on a tab that has no posts to show
2. ✅ Make sure that no results view is displayed correctly
3. Open a tab that has posts
4. While the poss are loading observe the ghost cells
5. ✅ Make sure that ghost cells are displayed correctly.
6. Observe a post cell after it's loaded.
7. ✅ Make sure that in the case where one avatar is shown, it is displayed correctly. 
8. ✅ Make sure that in the case where two avatars are shown, they are displayed correctly.
9. Scroll to the bottom.
8. ✅ Make sure the footer is displayed correctly.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
